pandas/core/construction.py

```diff
@@ -246,21 +253,25 @@ def array(
     """
     from pandas.core.arrays import (
         period_array,
+        IntegerArray,
         IntervalArray,
         PandasArray,
         DatetimeArray,
         TimedeltaArray,
+        StringArray,
     )
 
     if lib.is_scalar(data):
         msg = "Cannot pass scalar '{}' to 'pandas.array'."
         raise ValueError(msg.format(data))
 
-    data = extract_array(data, extract_numpy=True)
-
-    if dtype is None and isinstance(data, ABCExtensionArray):
+    if dtype is None and isinstance(
```

Comment:
The documentation stated that we use the `dtype` for Series, Index, and ExtensionArrays. Moved this up here to get the dtype from these before extracting.

This is important for the following:

```python
In [3]: s = pd.Series([1, 2])

In [4]: pd.array(s)
Out[4]:
<PandasArray>
[1, 2]
Length: 2, dtype: int64
```


Without this change, we would extract the data to an int64 ndarray. We ignore the dtype for ndarrays, so we would have `pd.array(pd.Series([1, 2]))` inferring `Int64Dtype()`, rather than NumPy's int64.